### ISSUE TITLE
Next Server Actions example does not work

### ISSUE THREAD

**Issue by rburgst** [TanStack/form#967]
### Describe the bug

The next server action example does not work currently

https://tanstack.com/form/latest/docs/framework/react/examples/next-server-actions

```
<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Can't resolve 'tunnel-agent' in '/home/projects/vbwimykwkg.github/node_modules/sharp'
<w> while resolving 'tunnel-agent' in /home/projects/vbwimykwkg.github/node_modules/sharp to a directory
 ✓ Compiled / in 5.9s (534 modules)
 GET / 200 in 6074ms
<w> [webpack.cache.PackFileCacheStrategy]
```

When submitting the form, the `useTransform` code blows up.

### Your minimal, reproducible example

https://stackblitz.com/github/tanstack/form/tree/main/examples/react/next-server-actions?embed=1&theme=dark&preset=node&file=src/index.tsx

### Steps to reproduce

1. Open the stackblitz
2. notice the sharp warning about `tunnel-agent`
3. submit the form
4. BOOM

### Expected behavior

The example should work and the server action should be invoked.

### How often does this bug happen?

Every time

### Screenshots or Videos

![Image](https://github.com/user-attachments/assets/7367fb64-5c79-4884-ad5c-88aeeff87354)

### Platform

- OS: Mac OSX Sequoia
- Browser: Arc

### TanStack Form adapter

None

### TanStack Form version

0.33.0

### TypeScript version

5.6.2

**Comment by rburgst** [comment]
In Stackblitz and codesandbox if I press the "Submit" button, I get
![Image](https://github.com/user-attachments/assets/dade41b4-13c3-4104-b803-022146e059d9)

**Comment by rburgst** [comment]
ok, I got it to work with the sample: 
https://stackblitz.com/edit/tanstack-form-bz8cz5?file=README.md

**Comment by anisharaz** [comment]
Am still getting the error and https://github.com/TanStack/form/issues/967#issuecomment-2387745999 this still have the issue , how did you get to work , what is the problem ?
![Image](https://github.com/user-attachments/assets/97af4e4c-bdfd-49a0-9b55-e33dd37da51e)

**Comment by rburgst** [comment]
@anisharaz see the PR #970

**Comment by rburgst** [comment]
Also for posterity: I have a next setup where also server side logging is shown in the browser console, however, for unknown reasons the logs from the server action never showed up in my browser console. They did show up in the server console though.
Also note that if you dont get your POST values on the action, then your `<input>` fields are missing the `name` attribute.

**Comment by anisharaz** [comment]
@rburgst Thankyou that fixed it , but a new problem arised or it exist am not sure, while submitting while the fields are empty the request goes to server and server validation invalidates the and responce with the message and client side validation run too. Does it makes sence to make to send request to server while the client validation is failing 
![Image](https://github.com/user-attachments/assets/f45c19b4-4633-4760-bbdb-d779eb74d196)
![Image](https://github.com/user-attachments/assets/2ccfa401-af54-4e88-b81f-d6158cad8695)

**Comment by rburgst** [comment]
@anisharaz this looks like a bug in tanstack form to me

**Comment by rburgst** [comment]
actually, this is due to the fact that if you did not touch the form state then "canSubmit" is true, only once you make any modification then the validation state is updated

**Comment by crutchcorn** [comment]
As far as I can tell, this was fixed in #970, please feel free to open a new issue if there's something I missed
